src/Compiler/Checking/AugmentWithHashCompare.fs

```diff
@@ -1333,7 +1359,22 @@ let MakeValsForEqualityWithComparerAugmentation g (tcref: TyconRef) =
     let withcEqualsVal =
         mkValSpec g tcref ty vis (Some(mkIStructuralEquatableEqualsSlotSig g)) "Equals" (tps +-> (mkEqualsWithComparerTy g ty)) tupArg false
 
-    objGetHashCodeVal, withcGetHashCodeVal, withcEqualsVal
+    let withcEqualsValExact =
+        mkValSpec 
+            g 
+            tcref 
+            ty
+            // Should be public no matter what.
+            // Otherwise generates internal Equals for internal DUs due to slot signature resolution.
+            (TAccess [])
```

Comment:
Can you reuse a static named + cached value for this?

Reply:
Not applicable anymore, keeping the visibility